frontend/src/components/common/CreateResourceButton.tsx

```diff
@@ -0,0 +1,210 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { getCluster } from '../../lib/cluster';
+import { apply } from '../../lib/k8s/apiProxy';
+import { KubeObjectInterface } from '../../lib/k8s/cluster';
+import { KubeConfigMap } from '../../lib/k8s/configMap';
+import { KubeRuntimeClass } from '../../lib/k8s/runtime';
+import { KubeSecret } from '../../lib/k8s/secret';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton, EditorDialog } from '../common';
+
+const creationTimestamp = new Date('2022-01-01').toISOString();
+const BASE_EMPTY_CONFIG_MAP: KubeConfigMap = {
```

Comment:
I think these templates should be part of each resource class.
Similar to what we are doing with https://github.com/headlamp-k8s/headlamp/pull/2036 .

We can have a `getBaseObject` static function. So we can do `ConfigMap.getBaseObject()` to get the very basic requirements for setting up such resources.